### DISCUSSION TITLE
pypa/pipenv vs pyenv/pyenv?

### DISCUSSION THREAD

**Discussion by oakla** [pypa/pipenv#5143] [Q&A]
I'm wondering whether there are any important differences between this repo (pypa/pipenv) and pyenv/pyenv.

After installing the other pyenv (pyenv/pyenv) via `git clone`, I spent a while getting my head around how it all works. Today I discoverd this version of pipenv (pypa) which I can install via pip.

1. Should I install pyenv via pip? 
2. Is there an important reason they're separate repos?
3. Are these two repos actually for the same tools, or is it just an amazing naming coincidence?

Sorry about what is probably a noob question.

Cheers

**Comment by matteius** [comment]
@oakla Did you mean to ask this in the `pypa/pyenv` repository?  There is even a different pyenv for when you are running windows ... `pyenv-win/pyenv-win`.   I forget which one I installed on linux, so maybe someone else can chime in about that.

**Reply by oakla** [reply]
Moments before you replied I realized that I misread 'pipenv' as 'pyenv' 🤦‍♂️. I'm pretty sure `pypa/pyenv` doesn't actually exist. I think I need to have a rest for the day. 

While I'm here, maybe someone can point me to the right tool to use for virtual environments with Jupyter?

Also. Should I just delete this thread?

**Reply by matteius** [reply]
You can leave it open @oakla 😄  -- it would be useful for others to chime in about what pyenv they are successfully using with pipenv and we can get to the bottom of the preferred versions.  I think there are some longer term goals around enhancing/adding to our documentation with OS specific items, and this kind of information could be useful as well.

**Reply by matteius** [reply]
I apparently have done a local install of `pyenv/pyenv` in linux, at least according to the `README.md` that links off to that repo.  Actually I see that `pypa/pyenv` doesn't exist, so yeah feel free to delete this if you want or mark this as answered.

**Reply by kalebmckale** [reply]
I'm a big fan of using the combination of `pyenv` and `pipenv` together.
